.github/workflows/bottest.yml

```diff
@@ -0,0 +1,127 @@
+    env:
+      PR_NUMBER: ${{ github.event.inputs.pr_number || github.event.issue.number || null }}
+    steps:
+      - name: Check membership
+        if: ${{ github.event_name == 'issue_comment' }}
+        env:
+          CONTRIB_ORG: clockworklabs
+          COMMENT_AUTHOR: ${{ github.event.comment.user.login }}
+          ORG_READ_TOKEN: ${{ secrets.ORG_READ_TOKEN }}
+        run: |
+          curl -OL https://github.com/cli/cli/releases/download/v2.37.0/gh_2.37.0_linux_amd64.deb && sudo dpkg -i gh_2.37.0_linux_amd64.deb
+          if [[ $(GH_TOKEN=$ORG_READ_TOKEN gh api --paginate /orgs/{owner}/members --jq 'any(.login == env.COMMENT_AUTHOR)') != true ]]; then
+            gh pr comment $PR_NUMBER -b "Sorry, you don't have permission to run bot tests."
+            exit 1
+          fi
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Post initial comment
+        run: |
+          if [[ $PR_NUMBER ]]; then
+            comment_parent=issues/$PR_NUMBER
+            comment_update=issues/comments
+          else
+            comment_parent=commits/$GITHUB_SHA
+            comment_update=comments
+          fi
+          comment_body="Bot test in progress..."
+          comment_id=$(gh api "/repos/{owner}/{repo}/$comment_parent/comments" -f body="$comment_body" --jq .id)
+          echo "COMMENT_UPDATE_URL=/repos/{owner}/{repo}/$comment_update/$comment_id" >>$GITHUB_ENV
+
+      - name: find PR branch
+        if: ${{ env.PR_NUMBER }}
+        run: echo "PR_REF=$(gh pr view $PR_NUMBER --json headRefName --jq .headRefName)" >>"$GITHUB_ENV"
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.PR_REF || github.ref }}
+          # if we're on master we want to know what the sha of HEAD~1 is so
+          # that we can compare results from it to HEAD (in the "Fetch markdown
+          # summary PR" step). otherwise, we can use a fully shallow checkout
+          fetch-depth: ${{ env.PR_NUMBER && 1 || 2 }}
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          components: clippy
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          override: true
+
+      - name: Built the bots
+        run: |
+          cargo build --release --bin spacetime
+
+      - name: Run bot test script
+        run: |
+          echo "We're on a bots runner, so expecting to find the BitCraftBots repo set up in the expected location"
+          echo "Let's hope the bots runner is up to date!!!"
+          SPACETIME_BIN="$PWD/target/release/spacetime"
+          pushd ~/BitCraftBots
+          python3 -m scripts.bottest "$SPACETIME_BIN"
+          popd
+          mkdir -p bottest-results
+          cp ~/BitCraftBots/results.zip bottest-results/$RESULTS_NAME.zip
+
+      - name: Upload bot test results to DO spaces
+        uses: shallwefootball/s3-upload-action@master
```

Comment:
I never know whether I should feel sketched out by using random GH actions like this (vs using the aws cli or something).
@cloutiertyler do you care?

Reply:
I mean, I don't love it. If it's open source, maybe we could just make our own copy?

Reply:
Made https://github.com/clockworklabs/s3-upload-action